File: median-array/median-array.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    const combinedArray = nums1.concat(nums2).sort(function(a,b){return a - b})
    const length = nums1.concat(nums2).sort().length
    const dividedlength = length % 2
    switch (dividedlength) {
        case 0:
            return (combinedArray[length / 2] + combinedArray[length / 2 - 1]) / 2
        default:
            return combinedArray[Math.floor(length / 2)]
    }
};

module.exports = { findMedianSortedArrays }

File: median-array/median-array.rb
# @param {Integer[]} nums1
# @param {Integer[]} nums2
# @return {Float}
def find_median_sorted_arrays(nums1, nums2)
    combined_array = nums1.concat(nums2).sort()
    length = nums1.concat(nums2).sort().length()
    divided_length = length % 2

    if divided_length == 0
        return ((combined_array[(length / 2)] + combined_array[(length / 2) - 1]) / 2)
    else
        return combined_array[(length / 2)].to_f
    end
end

print find_median_sorted_arrays([1, 2], [3, 4])